src/NuGet.Core/NuGet.Packaging/ContentModel/ReadOnlyMemoryCharComparerOrdinal.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Shared;
+
+namespace NuGet.ContentModel
+{
+    /// <summary>
+    /// Represents a comparer of <see cref="ReadOnlyMemory{T}" /> that uses string ordinal comparison.
+    /// </summary>
+    internal sealed class ReadOnlyMemoryCharComparerOrdinal : IEqualityComparer<ReadOnlyMemory<char>>
+    {
+        public static ReadOnlyMemoryCharComparerOrdinal Instance { get; } = new ReadOnlyMemoryCharComparerOrdinal();
+
+        private ReadOnlyMemoryCharComparerOrdinal()
+        {
+        }
+
+        public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
+        {
+            return x.Span.Equals(y.Span, StringComparison.Ordinal);
+        }
+
+        public int GetHashCode(ReadOnlyMemory<char> obj)
+        {
+            if (obj.Length == 0)
+            {
+                return 0;
+            }
+
+            var combiner = new HashCodeCombiner();
+            foreach (var character in obj.Span)
+            {
+                combiner.AddObject(character);
```

Comment:
Please use VS and "go to definition" (F12) on the AddObject method call.  I have a feeling it's going to use a version that causes boxing, and therefore causes memory allocations. Maybe changing this to `AddStruct` will avoid the allocations, but maybe we need a `char` overload.